altair/_magics.py

```diff
@@ -6,8 +6,28 @@
 import json
 import warnings
 
-import IPython
-from IPython.core import magic_arguments
+try:
+    import IPython
+    from IPython.core.magic_arguments import (magic_arguments, argument, parse_argstring)
+    IPYTHON_AVAILABLE=True
+except ImportError:
+    IPYTHON_AVAILABLE=False
+
+    # mocking the decorators if Ipython is unavailable
+    class magic_arguments:
+        def __init__(self, *args, **kwargs):
+            pass
+
+        def __call__(self, func):
+            pass
+
+    class argument:
+        def __init__(self, *args, **kwargs):
+            pass
+        def __call__(self, func):
+            pass
```

Comment:
Shouldn't ``__call__`` return ``func`` for this to be a no-op?

In any case, I don't think this all of extra logic is needed: ``_magic.py`` is only ever imported by the ``load_ipython_extension()`` function in ``__init__.py``, and this will only be invoked if IPython is installed (unless a user invokes/imports this deliberately, at which point I think an ``ImportError`` is appropriate).

Unless there's something I'm overlooking, I think all the changes to this file can be reverted.

Reply:
That was my first approach too. Unfortunately, as the tests are in the same folder structure as the code, pytest collects `_magic.py` which results in a failure because of the decorators if IPython is not installed. 
A very pragmatic solution would be to ignore that the tests fail if IPython is missing, as normally if the test requirements are installed, IPython is installed too...Is that ok?

Reply:
So essentially, what we would loose, is the ad hoc ability to e.g. test in in travis if everything works with and without IPython. But currently we don't do this anyhow, so for me this would be acceptable.

Reply:
And still, if we want to do that, we could tweak pytest config to ignore `_magic.py` from collecting. Not so nice, but would work somehow I guess.

Reply:
> pytest collects _magic.py which results in a failure because of the decorators if IPython is not installed.

Running pytest depends on the dev requirements, so putting IPython there should be fine, no?

Reply:
To be clear, what I'm saying is that ``_magic.py`` should be unchanged, and ``IPython`` should be a dev requirement, not an install requirement.

Reply:
OK, I see what you're saying.

As written, though, won't there be a ``NameError`` if pytest tries to import ``altair._magics`` with IPython not installed? I'd prefer an ``ImportError``, because it more directly tells the user what the problem is... and that's what you'd get if you leave this file unchanged.

I don't think pytest actually grabs the contents of this file unless you run ``--with-doctests``, in which case you need all the other dev requirements as well.